include/pagmo/problems/cec2009.hpp

```diff
@@ -0,0 +1,873 @@
+                // [0,1] x [-1,1]^{n-1}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                for (decltype(m_dim) i = 1u; i < m_dim; ++i) {
+                    lb[i] = -1.0;
+                    ub[i] = 1.0;
+                }
+            } else if (m_prob_id == 3u) {
+                // [0,1]^{n}
+                for (decltype(m_dim) i = 0u; i < m_dim; ++i) {
+                    lb[i] = 0.0;
+                    ub[i] = 1.0;
+                }
+            } else if (m_prob_id == 4u) {
+                // [0,1] x [-2,2]^{n-1}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                for (decltype(m_dim) i = 1u; i < m_dim; ++i) {
+                    lb[i] = -2.0;
+                    ub[i] = 2.0;
+                }
+            } else if (m_prob_id == 8u || m_prob_id == 9u || m_prob_id == 10u) {
+                // [0,1]^{2} x [-2,2]^{n-2}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                lb[1] = 0.0;
+                ub[1] = 1.0;
+                for (decltype(m_dim) i = 2u; i < m_dim; ++i) {
+                    lb[i] = -2.0;
+                    ub[i] = 2.0;
+                }
+            }
+        } else { // For CF
+            if (m_prob_id == 2u) {
+                // [0,1] x [-1,1]^{n-1}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                for (decltype(m_dim) i = 1u; i < m_dim; ++i) {
+                    lb[i] = -1.0;
+                    ub[i] = 1.0;
+                }
+            } else if (m_prob_id == 1u) {
+                // [0,1]^{n}
+                for (decltype(m_dim) i = 0u; i < m_dim; ++i) {
+                    lb[i] = 0.0;
+                    ub[i] = 1.0;
+                }
+            } else if (m_prob_id == 3u || m_prob_id == 4u || m_prob_id == 5u || m_prob_id == 6u || m_prob_id == 7u) {
+                // [0,1] x [-2,2]^{n-1}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                for (decltype(m_dim) i = 1u; i < m_dim; ++i) {
+                    lb[i] = -2.0;
+                    ub[i] = 2.0;
+                }
+            } else if (m_prob_id == 8u) {
+                // [0,1]^{2} x [-4,4]^{n-2}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                lb[1] = 0.0;
+                ub[1] = 1.0;
+                for (decltype(m_dim) i = 2u; i < m_dim; ++i) {
+                    lb[i] = -4.0;
+                    ub[i] = 4.0;
+                }
+            } else if (m_prob_id == 9u || m_prob_id == 10u) {
+                // [0,1]^{2} x [-2,2]^{n-2}
+                lb[0] = 0.0;
+                ub[0] = 1.0;
+                lb[1] = 0.0;
+                ub[1] = 1.0;
+                for (decltype(m_dim) i = 2u; i < m_dim; ++i) {
+                    lb[i] = -2.0;
+                    ub[i] = 2.0;
+                }
+            }
+        }
+        return std::pair<vector_double, vector_double>(std::move(lb), std::move(ub));
```

Comment:
If you want this can just be ``return std::make_pair(std::move(lb), std::move(ub));``.